openmc/mgxs/groups.py

```diff
@@ -0,0 +1,275 @@
+        if spacing == 'linear':
+            self.group_edges = np.linspace(start, stop, num_groups + 1)
+        elif spacing == 'logarithmic':
+            self.group_edges = \
+                np.logspace(np.log10(start), np.log10(stop), num_groups + 1)
+
+        self._num_groups = num_groups
+
+    def get_group(self, energy):
+        """Returns the energy group in which the given energy resides.
+
+        Parameters
+        ----------
+        energy : Real
+            The energy of interest in MeV
+
+        Returns
+        -------
+        Integral
+            The energy group index, starting at 1 for the highest energies
+
+        Raises
+        ------
+        ValueError
+            If the group edges have not yet been set.
+
+        """
+
+        if self.group_edges is None:
+            msg = 'Unable to get energy group for energy "{0}" MeV since ' \
+                  'the group edges have not yet been set'.format(energy)
+            raise ValueError(msg)
+
+        index = np.where(self.group_edges > energy)[0]
+        group = self.num_groups - index
+        return group
+
+    def get_group_bounds(self, group):
+        """Returns the energy boundaries for the energy group of interest.
+
+        Parameters
+        ----------
+        group : Integral
+            The energy group index, starting at 1 for the highest energies
+
+        Returns
+        -------
+        2-tuple
+            The low and high energy bounds for the group in MeV
+
+        Raises
+        ------
+        ValueError
+            If the group edges have not yet been set.
+
+        """
+
+        if self.group_edges is None:
+            msg = 'Unable to get energy group bounds for group "{0}" since ' \
+                  'the group edges have not yet been set'.format(group)
+            raise ValueError(msg)
+
+        lower = self.group_edges[self.num_groups-group]
+        upper = self.group_edges[self.num_groups-group+1]
+        return lower, upper
+
+    def get_group_indices(self, groups='all'):
+        """Returns the array indices for one or more energy groups.
+
+        Parameters
+        ----------
+        groups : str, tuple
+            The energy groups of interest - a tuple of the energy group indices,
+            starting at 1 for the highest energies (default is 'all')
+
+        Returns
+        -------
+        ndarray
+            The ndarray array indices for each energy group of interest
+
+        Raises
+        ------
+        ValueError
+            If the group edges have not yet been set, or if a group is requested
+            that is outside the bounds of the number of energy groups.
+
+        """
+
+        if self.group_edges is None:
+            msg = 'Unable to get energy group indices for groups "{0}" since ' \
+                  'the group edges have not yet been set'.format(groups)
+            raise ValueError(msg)
+
+        if groups == 'all':
+            indices = np.arange(self.num_groups)
```

Comment:
Yes, this is something that I (clearly) did not adequately test, but will include it in one of my tests for this module in the suite.